### PR TITLE
Moving ships

### PR DESCRIPTION
**Pull request by Crollt**

### CODE REVIEW

File: src/test/java/cs361/battleships/models/BoardTest.java [hunk 1]
```diff
@@ -102,4 +100,28 @@ public void testGameEnd(){
...
+        Result res = new Result();
+        res.setResult(AtackStatus.SUNK);
+        atks.add(res);
+        atks.add(res);
+        board.setAttacks(atks);
+        board.placeShip(new Minesweeper(), 5, 'C', false); //place all 3 ships
+        board.placeShip(new Submarine(), 2, 'C', false); //place all 3 ships
+        assertTrue(board.move(0, 2));
+        assertTrue(board.move(1, 2));
+        assertTrue(board.move(2, 2));
```
[the remark this answers is not in the document]
Comment (Crollt) [reply src/test/java/cs361/battleships/models/BoardTest.java]: Returns a boolean, the game.js checks the number of moves left, so it doesn't matter how many times you move it in here.

File: src/main/java/cs361/battleships/models/Board.java [hunk 2]
```diff
@@ -290,6 +290,191 @@ private boolean checkSonarCondition() {
...
+		this.ships = shipMoves;
+		return true;
+	}
+
+	private boolean checkMoveCondition() {
+		int numSunk = 0;
+		for (Result r : this.attacks) {
+			if (r.getResult() == AtackStatus.SUNK) {
+			    numSunk++;
+			}
```
[the remark this answers is not in the document]
Comment (Crollt) [reply src/main/java/cs361/battleships/models/Board.java]: Thanks.

File: src/test/java/cs361/battleships/models/BoardTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hartwelg) [reply src/test/java/cs361/battleships/models/BoardTest.java]: looks good to me as well

File: src/main/java/cs361/battleships/models/Board.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hartwelg) [reply src/main/java/cs361/battleships/models/Board.java]: Ships do interact correctly, good job

**Pull request merged**